DiverseSelector/metric.py

```diff
@@ -23,25 +23,292 @@
 
 """Metric calculation module."""
 
+from typing import Any
+
 import numpy as np
-from scipy.spatial.distance import cdist
+from scipy.spatial.distance import cdist, squareform
+from sklearn.metrics import pairwise_distances
+
+from DiverseSelector.utils import sklearn_supported_metrics
 
 __all__ = [
     "pairwise_dist",
     "compute_diversity",
+    "distance_to_similarity",
+    "pairwise_similarity",
+    "pairwise_similarity_bit",
+    "tanimoto",
+    "cosine",
+    "dice",
+    "bit_tanimoto",
+    "bit_cosine",
+    "bit_dice",
 ]
 
 
+class ComputeDistanceMatrix:
```

Comment:
This class is a wrapper for all other functions to compute distances. For each function, the input would be just the feature matrix (num_molecules * num_features) and the features can either be molecular descriptors or fingerprints (bit matrix). Once all the distance functions are finished, we can finish this Python class here.